modules/serverless-nodejs-functions-context-objects.adoc

```diff
@@ -0,0 +1,57 @@
+// Module included in the following assemblies
+//
+// * /serverless/functions/serverless-developing-nodejs-functions.adoc
+
+[id="serverless-nodejs-functions-context-objects_{context}"]
+= Node.js context objects
+
+Functions are invoked by providing a `context` object as the first parameter.
+
+.Example context object
+[source,js]
+----
+function processPurchase(context, data)
+----
+
+This object provides access to the incoming HTTP request information, including the HTTP request method, any query strings or headers sent with the request, the HTTP version, and the request body. Incoming requests that contain a CloudEvent attach the incoming instance of the CloudEvent to the context object, so that it can be accessed by using `context.cloudevent`.
+
+[id="serverless-nodejs-functions-context-objects-methods_{context}"]
+== Context object methods
+
+The `context` object has a single method, `cloudEventResponse()` that accepts a data value and returns a CloudEvent.
+
+In a Knative system, if a function deployed as a service is invoked by an event broker sending a CloudEvent, the broker will examine the response. If the response is a CloudEvent, this event will then be handled by the broker.
+
+.Example context object method
+[source,js]
+----
+// Expects to receive a CloudEvent with customer data
+function processCustomer(context, customer) {
+  // process the customer
+  const processed = processCustomer(customer);
+  return context.cloudEventResponse(customer);
```

Comment:
```suggestion
  return context.cloudEventResponse(customer)
    .source('/processCustomer')
    .type('fn.process.customer')
    .response();
```

This is actually a better example, showing customization of the source and type of return event.